compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -105,12 +108,22 @@ buildDepClassMap :: Config -> DepClassMap
 buildDepClassMap Config{..} = DepClassMap $ MS.fromList
     [ ((moduleName, synName), (packageId, dsyn))
     | packageId <- Set.toList configDependencyPackages
-    , let unitId = configGetUnitId (LF.PRImport packageId)
-    , Just LF.Package{..} <- [MS.lookup unitId configPackages]
+    , Just LF.Package{..} <- [MS.lookup packageId configPackages]
     , LF.Module{..} <- NM.toList packageModules
     , dsyn@LF.DefTypeSyn{..} <- NM.toList moduleSynonyms
     ]
 
+buildDepInstances :: Config -> MS.Map LF.TypeSynName [LF.Type]
+buildDepInstances Config{..} = MS.fromListWith (<>)
+    [ (clsName, [snd dvalBinder])
+    | packageId <- Set.toList configDependencyPackages
+    , Just LF.Package{..} <- [MS.lookup packageId configPackages]
+    , LF.Module{..} <- NM.toList packageModules
+    , LF.DefValue{..} <- NM.toList moduleValues
+    , Just dfun <- [getDFunSig dvalBinder]
+    , let clsName = LF.qualObject $ dfhName $ dfsHead dfun
```

Comment:
I see, this map conflates different classes with the same name, but it's ok because this is only an optimization and the real collision check in `isDuplicate` tests that the types are equal. It might be nice to add a comment.

Reply:
Good idea, I’ve added a comment.